core/domain/user_jobs_one_off.py

```diff
@@ -17,348 +17,11 @@
-        created_exploration_ids = set()
-        edited_exploration_ids = set()
-
-        edits = [ast.literal_eval(v) for v in version_and_exp_ids]
-        for edit in edits:
-            edited_exploration_ids.add(edit['exploration_id'])
-            if edit['version_string'] == '1':
-                created_exploration_ids.add(edit['exploration_id'])
-
-        if user_services.get_user_contributions(key, strict=False) is not None:
-            user_services.update_user_contributions(
-                key, list(created_exploration_ids), list(
-                    edited_exploration_ids))
-        else:
-            user_services.create_user_contributions(
-                key, list(created_exploration_ids), list(
-                    edited_exploration_ids))
-
-
-class UsernameLengthDistributionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
-    """One-off job for calculating the distribution of username lengths."""
-
-    @classmethod
-    def entity_classes_to_map_over(cls):
-        """Return a list of datastore class references to map over."""
-        return [user_models.UserSettingsModel]
-
-    @staticmethod
-    def map(item):
-        """Implements the map function for this job."""
-        if item.username is not None:
-            yield (len(item.username), 1)
-
-    @staticmethod
-    def reduce(key, stringified_username_counter):
-        """Implements the reduce function for this job."""
-        username_counter = [
-            ast.literal_eval(v) for v in stringified_username_counter]
-        yield (key, len(username_counter))
-
-
-class UsernameLengthAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
-    """Job that audits and validates username lengths."""
-
-    @classmethod
-    def entity_classes_to_map_over(cls):
-        return [user_models.UserSettingsModel]
-
-    @staticmethod
-    def map(model_instance):
-        if len(model_instance.username) > 20:
-            yield (len(model_instance.username), model_instance.username)
-
-    @staticmethod
-    def reduce(key, values):
-        yield ('Length: %s' % key, 'Usernames: %s' % sorted(values))
-
-
-class LongUserBiosOneOffJob(jobs.BaseMapReduceOneOffJobManager):
-    """One-off job for calculating the length of user_bios."""
-
-    @classmethod
-    def entity_classes_to_map_over(cls):
-        """Return a list of datastore class references to map over."""
-        return [user_models.UserSettingsModel]
-
-    @staticmethod
-    def map(item):
-        """Implements the map function for this job."""
-        if item.user_bio is None:
-            user_bio_length = 0
-        else:
-            user_bio_length = len(item.user_bio)
-
-        yield (user_bio_length, item.username)
-
-    @staticmethod
-    def reduce(userbio_length, stringified_usernames):
-        """Implements the reduce function for this job."""
-        if int(userbio_length) > 500:
-            yield (userbio_length, stringified_usernames)
-
-
-class PopulateStoriesAndTopicsOneOffJob(
```

Comment:
This job is required and a part of Milestone 1 of @krishita30j's GSoC project, so this should be kept. It is already tested on backup server and to be run in June release.
Can removal of this be shifted to after the June release is deployed to prod so as to fix any issues on the job, if any.

Reply:
Unfortunately not, this PR (and the rest of the Python 3 migration) needs to go in ASAP and is targeted for the July release. Please have them write an Apache Beam job instead, I plan on posting a Wiki Page tomorrow.